docs/tutorials/realistic_sims/noise.md

```diff
@@ -1,7 +1,37 @@
 Running programs on NISQ devices often leads to partially useful results due to the presence of noise.
-In order to perform realistic simulations, a number of noise models are supported in Qadence and
+In order to perform realistic simulations, a number of noise models (for digital operations, analog operations and simulated readout errors) are supported in Qadence and
 corresponding error mitigation techniques whenever possible.
 
+# NoiseHandler
+
+Noise models can be defined via the `NoiseHandler`. It is a container of several `NoiseSource` instances which require to specify a `protocol`,
+a dictionary of `options`. The `protocol` field is to be instantiated from `NoiseProtocol`.
+
+```python exec="on" source="material-block" session="noise" result="json"
+from qadence import NoiseHandler, NoiseSource
+from qadence.types import NoiseProtocol
+
+analog_noise = NoiseHandler(protocol=NoiseProtocol.ANALOG.DEPOLARIZING, options={"noise_probs": 0.1})
+digital_noise = NoiseHandler(protocol=NoiseProtocol.DIGITAL.DEPOLARIZING, options={"error_probability": 0.1})
+readout_noise = NoiseHandler(protocol=NoiseProtocol.READOUT.READOUT, options={"error_probability": 0.1, "seed": 0})
+```
+
+One can also define a `NoiseHandler` as a list of `NoiseSource` instances:
+
+```python exec="on" source="material-block" session="noise" result="json"
+from qadence import NoiseHandler, NoiseSource
+from qadence.types import NoiseProtocol
+
+digital_noise = NoiseSource(protocol=NoiseProtocol.DIGITAL.DEPOLARIZING, options={"error_probability": 0.1})
+readout_noise = NoiseSource(protocol=NoiseProtocol.READOUT.READOUT, options={"error_probability": 0.1, "seed": 0})
+
+noise_combination = NoiseHandler([digital_noise, readout_noise])
```

Comment:
If the `NoiseHandler` can accept a single protocol (as shown in line 61 below) why would it need a list of `NoiseSource` ? It feels a bit on an unnecessary intermediate step unless there is a reason I don't see now.

Reply:
This is especially for the case of Digital noise where we can combine several types (if the user decide to apply let's say a bitflip followed by a phase flip, etc)

Reply:
Gotcha, now it's been reworked.